src/eth/types.test-d.ts

```diff
@@ -0,0 +1,85 @@
+import { expectAssignable, expectNotAssignable } from 'tsd';
```

Comment:
We were talking about the name of the file previously, looking at tsd documentation this seems to be the correct naming so I think we should update our test configuration.

https://github.com/tsdjs/tsd

Reply:
After discussing with @danroc, I was able to make it work without... any changes 🙈

IDK what I did yesterday during my testing, but it works actually fine, it's just that `tsd` is very silent and does output something only when there are errors!

So nothing to change here! :)